### PR TITLE
Handle the case with no queries in `GPUStatsMonitor`

### PR DESCRIPTION
**Pull request by michele-arrival**
Call to nvidia-smi in GPUStatsMonitor would fail if there are no queries. GPUStatsMonitor can be still used to log `batch_time/intra_step (ms)` and `batch_time/inter_step (ms)`. Skipping the call to nvidia-smi also avoids unnecessary overhead.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/callbacks/test_gpu_stats_monitor.py [hunk 1]
```diff
@@ -63,6 +63,37 @@ def test_gpu_stats_monitor(tmpdir):
...
+        temperature=False,
+    )
+    logger = CSVLogger(tmpdir)
+    log_every_n_steps = 2
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=2,
+        limit_train_batches=7,
+        log_every_n_steps=log_every_n_steps,
```
Comment (carmocca) [inline tests/callbacks/test_gpu_stats_monitor.py:87]: Running 2 batches can be enough here

```suggestion
        fast_dev_run=2
```

File: tests/callbacks/test_gpu_stats_monitor.py [hunk 2]
```diff
@@ -63,6 +63,37 @@ def test_gpu_stats_monitor(tmpdir):
...
+        gpus=1,
+        callbacks=[gpu_stats],
+        logger=logger,
+    )
+
+    trainer.fit(model)
+    assert trainer.state.finished, f"Training failed with {trainer.state}"
```
Comment (carmocca) [inline tests/callbacks/test_gpu_stats_monitor.py:94]: Can you add assertions for the `CSVLogger` state?

Comment (michele-arrival) [reply tests/callbacks/test_gpu_stats_monitor.py:94]: same as the test just above (i.e. test_gpu_stats_monitor? )

Comment (michele-arrival) [reply tests/callbacks/test_gpu_stats_monitor.py:94]: minus the `fields = ["utilization.gpu", "memory.used", "memory.free", "utilization.memory"]`

Comment (michele-arrival) [reply tests/callbacks/test_gpu_stats_monitor.py:94]: i guess see 0579717755a97ece28abf10dd21341683ae4b03c if it's ok?

Comment (carmocca) [reply tests/callbacks/test_gpu_stats_monitor.py:94]: Actually, I would avoid using the `CSVLogger` entirely and `unittest.mock.patch` the `trainer.logger.log_metrics` method which is called by the `GPUStatsMonitor` 

https://github.com/PyTorchLightning/pytorch-lightning/blob/ad3f183bc367095b3f121882489fb532983c672a/pytorch_lightning/callbacks/gpu_stats_monitor.py#L179

Are you familiar with mocking and asserting its calls? Ping if you need any help

File: tests/callbacks/test_gpu_stats_monitor.py (hunk 1 above)
Comment (michele-arrival) [reply tests/callbacks/test_gpu_stats_monitor.py]: done

File: tests/callbacks/test_gpu_stats_monitor.py (hunk 2 above)
Comment (michele-arrival) [reply tests/callbacks/test_gpu_stats_monitor.py:94]: not exactly, but i can get up to speed

Comment (michele-arrival) [reply tests/callbacks/test_gpu_stats_monitor.py:94]: is this what you had in mind?

File: pytorch_lightning/callbacks/gpu_stats_monitor.py [hunk 3]
```diff
@@ -187,6 +187,9 @@ def _get_gpu_ids(device_ids: List[int]) -> List[str]:
         return [cuda_visible_devices[device_id].strip() for device_id in device_ids]
 
     def _get_gpu_stats(self, queries: List[str]) -> List[List[float]]:
+        if len(queries) == 0:
```
Comment (carmocca) [inline pytorch_lightning/callbacks/gpu_stats_monitor.py:190]:
```suggestion
        if not queries:
```

**Review by carmocca — APPROVED** [review]

File: tests/callbacks/test_gpu_stats_monitor.py (hunk 2 above)
Comment (carmocca) [reply tests/callbacks/test_gpu_stats_monitor.py]: Pushed https://github.com/PyTorchLightning/pytorch-lightning/pull/9014/commits/fe7416ccb8e918ee4fbe48e1debfeb74c4c1db2b with what I had in mind 👍

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/callbacks/gpu_stats_monitor.py [hunk 4]
```diff
@@ -187,6 +187,9 @@ def _get_gpu_ids(device_ids: List[int]) -> List[str]:
         return [cuda_visible_devices[device_id].strip() for device_id in device_ids]
 
     def _get_gpu_stats(self, queries: List[str]) -> List[List[float]]:
+        if not queries:
```
Comment (ananthsub) [inline pytorch_lightning/callbacks/gpu_stats_monitor.py:190]: super n00b question: is `not queries` for truthiness preferred over `if len(queries) == 0` ?

**Review by ananthsub — APPROVED** [review]
